### ISSUE TITLE
Delayed PC shutdown due to Keeweb don't shutdown correctly

### ISSUE THREAD

**Issue by elbowz** [keeweb/keeweb#1773]
**Describe the bug**
During the computer shutdown Keeweb delay the power off, I think because don't close correctly (SIGHUP) and it must be force closed (SIGKILL)

**To Reproduce**
1. start keeweb
(no need fill pwd)
2. normal shutdown the pc

**Environment**
KeeWeb v1.17.5 (4ee87e19, 2021-03-27)
Environment: electron v11.3.0
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) KeeWeb/1.17.5 Chrome/87.0.4280.141 Electron/11.3.0 Safari/537.36

**Kdbx File**
Database on GDrive

**Logs**
```
journalctl -b -1
apr 01 15:47:51 xps systemd[1]: Stopped User Login Management.
apr 01 15:47:51 xps audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=systemd-logind comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
apr 01 15:47:51 xps audit: BPF prog-id=20 op=UNLOAD
apr 01 15:47:51 xps audit: BPF prog-id=19 op=UNLOAD
apr 01 15:49:18 xps systemd[677]: app-keeweb-5ad58df86817436792affeb44f4b21ba.scope: Stopping timed out. Killing.
apr 01 15:49:18 xps kernel: kauditd_printk_skb: 13 callbacks suppressed
apr 01 15:49:18 xps systemd[677]: app-keeweb-5ad58df86817436792affeb44f4b21ba.scope: Stopping timed out. Killing.
apr 01 15:49:18 xps kernel: kauditd_printk_skb: 13 callbacks suppressed
apr 01 15:49:18 xps kernel: audit: type=1131 audit(1617284958.962:161): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=user@1000 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
apr 01 15:49:18 xps kernel: audit: type=1131 audit(1617284958.979:162): pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=user-runtime-dir@1000 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
apr 01 15:49:18 xps audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=user@1000 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
apr 01 15:49:18 xps audit[1]: SERVICE_STOP pid=1 uid=0 auid=4294967295 ses=4294967295 msg='unit=user-runtime-dir@1000 comm="systemd" exe="/usr/lib/systemd/systemd" hostname=? addr=? terminal=? res=success'
apr 01 15:49:18 xps systemd[678]: pam_warn(systemd-user:setcred): function=[pam_sm_setcred] flags=0x8004 service=[systemd-user] terminal=[] user=[muttley] ruser=[<unknown>] rhost=[<unknown>]
apr 01 15:49:18 xps systemd[677]: app-keeweb-5ad58df86817436792affeb44f4b21ba.scope: Killing process 1901 (electron) with signal SIGKILL.
apr 01 15:49:18 xps systemd[677]: app-keeweb-5ad58df86817436792affeb44f4b21ba.scope: Killing process 1935 (n/a) with signal SIGKILL.
apr 01 15:49:18 xps systemd[677]: app-keeweb-5ad58df86817436792affeb44f4b21ba.scope: Killing process 1938 (ThreadPoolForeg) with signal SIGKILL.
apr 01 15:49:18 xps systemd[677]: app-keeweb-5ad58df86817436792affeb44f4b21ba.scope: Killing process 1939 (inotify_reader) with signal SIGKILL.
apr 01 15:49:18 xps systemd[677]: app-keeweb-5ad58df86817436792affeb44f4b21ba.scope: Killing process 1974 (ThreadPoolSingl) with signal SIGKILL.
apr 01 15:49:18 xps systemd[677]: app-keeweb-5ad58df86817436792affeb44f4b21ba.scope: Failed with result 'timeout'.
```

**Notes**
You can see the gap time (15:47:51 - 15:49:18) waiting Keeweb closing.
Is there a way to save keeweb log in a file?

p.s. sorry to boring you with another issue ;)

**Comment by antelle** [comment]
Interesting, on macOS it exits on `SIGHUP`. KeeWeb is just opened, right? There's for example no database open in KeeWeb that would prevent it from closing.

**Comment by antelle** [comment]
Tested in on Ubuntu, seems to be working as expected. Which distributive did you use and how KeeWeb was launched?

**Comment by elbowz** [comment]
TL;DR for sure my platform...sorry for wasted time

Arch linux 5.11.11
Installed keeweb by AUR package: https://aur.archlinux.org/packages/keeweb/

I have also tested on other pc with Ubuntu without trouble...so, some issue/regression introduced by the rolling release philosophy under Arch (or something other black magic).

I'm quite sure this will be fixed in the next update of systemd, electron, etc...

p.s. if there was a way to keep a file log, I was curiously what happen to keeweb when is reached to kill signal during the shutdown phase.

Thx!

**Comment by antelle** [comment]
I wonder if it's something that happens only in 5.11.11. I have Arch linux 5.11.6 with KDE and it also works there:

![image](https://user-images.githubusercontent.com/633557/113409475-a3b56280-93b1-11eb-9d6b-1eff8205a341.png)

Showdown / logoff also works almost immediately with running KeeWeb.

Do you have any setting changed in KeeWeb? For example, minimize instead of close can cause this effect.

Regarding logs, they can contain private information, so there's no way to save them. We should add an option for troubleshooting such cases.

**Comment by elbowz** [comment]
I'm come back to home, where there is the arch linux pc with the trouble...and the trouble is still there :(

I have made a sys update, and some packages are been update, among these:
* updated electron (11.3.0-1 -> 12.0.2-1)
* installed electron11 (11.4.1-1)
* updated systemd-libs (247.4-2 -> 248-1)

also forced a reinstall/repackage keeweb from source (AUR repository)...and now it works like a charm :+1:

**Comment by antelle** [comment]
Okay, good to hear it works!
